tick/stateful/eval_function_node.go

```diff
@@ -31,16 +32,36 @@ func NewEvalFunctionNode(funcNode *ast.FunctionNode) (*EvalFunctionNode, error)
 	return evalFuncNode, nil
 }
 
-func (n *EvalFunctionNode) Type(scope ReadOnlyScope, executionState ExecutionState) (ast.ValueType, error) {
-	// PERF: today we are evaluating the function, it will be much faster if will type info the function it self
-	result, err := n.callFunction(scope.(*Scope), executionState)
-	if err != nil {
-		return ast.InvalidType, err
+func (n *EvalFunctionNode) Type(scope ReadOnlyScope) (ast.ValueType, error) {
+	f := GlobalExecutionSate.Funcs[n.funcName]
+	if f == nil {
+		df := scope.DynamicFunc(n.funcName)
+		if df == nil {
+			return ast.InvalidType, fmt.Errorf("undefined function: %q", n.funcName)
+		}
+		f = df
 	}
+	signature := f.Signature()
 
-	// We can't cache here the result (although it's very tempting ;))
-	// because can't trust function to return always the same consistent type
-	return ast.TypeOf(result), nil
+	domain := Domain{}
+	if gotLen, expLen := len(n.argsEvaluators), len(domain); gotLen > expLen {
+		return ast.InvalidType, fmt.Errorf("functions do not accept more than %v arguements: received %v", expLen, gotLen)
```

Comment:
Would it be better to return an ErrWrongFuncSignature here instead so the user has an idea of how to fix the problem?